app/viewmodels/DutyDefermentAccountViewModel.scala

```diff
@@ -0,0 +1,253 @@
+            msg = messages("cf.account.details.previous-eori", statement.eoriHistory.eori)))
+        } else {
+          None
+        }
+
+        val headWithEntriesOrNoStatement = if (statement.groups.head.monthAndYear.compareTo(monthsToDisplay) > 0) {
+          val ddHead = new duty_deferment_head(new h2(), new p()).apply(statement.groups.head)
+
+          val entries: Seq[HtmlFormat.Appendable] = statement.groups.tail.map {
+            entry =>
+              if (entry.monthAndYear.compareTo(monthsToDisplay) > 0) {
+                new duty_deferment_accordian(new GovukAccordion()).apply(Seq(entry), 0)
+              } else {
+                new duty_deferment_accordian(new GovukAccordion()).apply(Seq(), 0)
+              }
+          }
+
+          DDHeadWithEntriesOrNoStatements(ddHeadWithEntry = Some(DDHeadWithEntry(ddHead, entries)))
+        } else {
+          DDHeadWithEntriesOrNoStatements(
+            noStatementsMsg =
+              Some(new inset().apply(msg = messages("cf.account.detail.no-statements", accountNumber))))
+        }
+
+        Some(FirstPopulatedStatement(historicEoriHeading, headWithEntriesOrNoStatement))
+      }
+    }
+  }
+
+  private def prepareTailingStatements(statements: Seq[DutyDefermentStatementsForEori])
+                                      (implicit messages: Messages): Seq[TailingStatement] = {
+    statements.zipWithIndex.map {
+      statement =>
+        val statementsForEori = statement._1
+        val historyIndex = statement._2
+
+        val historicEoriHeading =
+          if (statementsForEori.eoriHistory.isHistoricEori && statementsForEori.currentStatements.nonEmpty) {
+            Some(new h2().apply(
+              id = Some(s"historic-eori-${historyIndex + 1}"),
+              msg = messages("cf.account.details.previous-eori", statementsForEori.eoriHistory.eori)
+            ))
+          } else {
+            None
+          }
+
+        val accordian = new duty_deferment_accordian(new GovukAccordion())
+          .apply(statementsForEori.groups, historyIndex + 1)
+
+        TailingStatement(historicEoriHeading, accordian)
+    }
+  }
+
+  private def statOlderThanSixMonths(serviceUnavailableUrl: String)(implicit messages: Messages): GuidanceRow = {
+    GuidanceRow(h2Heading = new h2().apply(
+      id = Some("missing-documents-guidance-heading"),
+      msg = messages("cf.common.missing-documents-guidance.cdsStatements.heading")),
+
+      link = Some(new link().apply(
+        pId = Some("missing-documents-guidance-text1"),
+        linkMessage = "cf.accounts.older-statements.description.link",
+        location = serviceUnavailableUrl,
+        linkClass = "govuk-link govuk-link--no-visited-state",
+        preLinkMessage = Some("cf.accounts.older-statements.description")
+      )))
+  }
+
+  private def chiefDeclaration(implicit appConfig: AppConfig,
+                               messages: Messages): GuidanceRow = {
+    GuidanceRow(h2Heading = new h2().apply(
```

Comment:
There is a bit of code duplication in the way elements are being created `(h2, p, link, etc.)`. Maybe consider creating utility methods to reduce this duplication.

Reply:
Utility methods have been created to populate the components